juju/juju
# upgrade-charm, updates the charm's lxd profile too

## Description of change

During charm upgrade steps: update the lxd profile from the charm for an lxd machine; add an lxd profile to an lxd machine if profile new to charm.

## QA steps
`export JUJU_DEV_FEATURE_FLAGS=lxd-profile`
1. `juju bootstrap localhost --build-agent`
2. `juju deploy ./testcharms/charm-repo/quantal/lxd-profile lxd-profile-local`
3. `juju add-unit  lxd-profile-local --to lxd`
3. notice `lxc profile juju-default-lxd-profile-local-0` created and used
4. `juju upgrade-charm lxd-profile-local --path  ./testcharms/charm-repo/quantal/lxd-profile`
5. notice `lxc profile juju-default-lxd-profile-local-0` deleted; `juju-default-lxd-profile-local-0` created and now used.
6. `rm ./testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml`
7. `juju deploy ./testcharms/charm-repo/quantal/lxd-profile lxd-local`
8. notice `lxc profile juju-default-lxd-local-0` does not exist
9. replace `./testcharms/charm-repo/quantal/lxd-profile/lxd-profile.yaml`
10. `juju upgrade-charm lxd-local --path  ./testcharms/charm-repo/quantal/lxd-profile`
8. notice `lxc profile juju-default-lxd-local-1` exists and is used.
